monocle/webapp.py

```diff
@@ -121,11 +121,37 @@ def whoami():
     return jsonify(username)
 
 
-@app.route("/api/0/query/<name>", methods=["GET"])
-def query(name):
+def _get_index(request):
     if not request.args.get("index"):
         abort(make_response(jsonify(errors=["No index provided"]), 404))
-    index = request.args.get("index")
+    return request.args.get("index")
+
+
+def _get_project_definition(index):
```

Comment:
You should read the ProjectConfig the same way than in line 81. And put the data in global (yes that's bad, there is a single place on that file where we read the configuration file so better to not duplicate).

Reply:
I really don't like. Putting something as a global is going to raise some problems, but as you prefer.